Application/EdFi.Ods.Api/Security/AuthorizationStrategies/CustomViewBased/CustomViewBasisEntityProvider.cs

```diff
@@ -0,0 +1,58 @@
+using System.Collections.Concurrent;
+using System.Linq;
+using EdFi.Ods.Common.Conventions;
+using EdFi.Ods.Common.Models;
+using EdFi.Ods.Common.Models.Domain;
+
+namespace EdFi.Ods.Api.Security.AuthorizationStrategies.CustomViewBased;
+
+public class CustomViewBasisEntityProvider : ICustomViewBasisEntityProvider
```

Comment:
To avoid the need for security metadata database changes, we use an authorization strategy naming convention for custom view-based authorization strategies. It uses the following format: `{entityName}With...`. The entity name is used to find a matching entity in the Ed-Fi Standard and failing that it looks in extension schemas (in no particular order) until it finds a matching entity. The primary key of that entity then is deemed to be the basis for joining with the authorization subject (the focus of the current requested resource).